src/modules/tests/test-data/cat-be/test-data.cat-be.selector.ts

```diff
@@ -0,0 +1,119 @@
+  data: CatBEUniqueTypes.TestRequirements, legalRequirement: LegalRequirements) => {
+  return data[legalRequirement];
+};
+
+export const getVehicleChecks = (
+  state: CatBEUniqueTypes.TestData): CatBEUniqueTypes.VehicleChecks => state.vehicleChecks;
+
+export const getTellMeQuestion = (state: CatBEUniqueTypes.VehicleChecks): VehicleChecksQuestion => {
+  const questionProvider: QuestionProvider =  new QuestionProvider();
+  return questionProvider
+    .getTellMeQuestions(TestCategory.BE)
+    .find(question => question.code === get(state, 'tellMeQuestion.code'));
+};
+
+export const areTellMeQuestionsSelected = (
+  state: CatBEUniqueTypes.VehicleChecks) => typeof get(state, 'tellMeQuestions') !== 'undefined';
+
+export const areTellMeQuestionsCorrect = (state: CatBEUniqueTypes.VehicleChecks) => {
+  const tellMeQuestions = get(state, 'tellMeQuestions');
+  let correct = true;
+
+  if (typeof tellMeQuestions === 'undefined' || tellMeQuestions === null || !(tellMeQuestions instanceof Array)) {
+    correct = false;
+  } else {
+    tellMeQuestions.forEach((question) => {
+      if (question.outcome !== CompetencyOutcome.P) {
+        correct = false;
+      }
+    });
+  }
+
+  return correct;
+};
+
+export const getCatBELegalRequirements = (data: CatBEUniqueTypes.TestData): CatBELegalRequirements => {
+  return {
+    normalStart1: data.testRequirements.normalStart1 || false,
+    normalStart2: data.testRequirements.normalStart2 || false,
+    downhillStart: data.testRequirements.downhillStart || false,
+    uphillStart: data.testRequirements.uphillStart || false,
+    angledStartControlledStop: data.testRequirements.angledStartControlledStop || false,
+    manoeuvre: hasManoeuvreBeenCompleted(data) || false,
+    vehicleChecks: hasVehicleChecksBeenCompleted(data) || false,
+    eco: data.eco.completed || false,
+  };
+};
+
+export const hasVehicleChecksBeenCompleted = (data: CatBEUniqueTypes.TestData): boolean => {
+  const REQUIRED_SHOW_ME_QUESTIONS = 3;
+  const REQUIRED_TELL_ME_QUESTIONS = 2;
+
+  let showMeQuestionComplete = true;
+  let tellMeQuestionComplete = true;
+
+  if (
+    typeof data.vehicleChecks.showMeQuestions === 'undefined' ||
+    data.vehicleChecks.showMeQuestions === null ||
+    !(data.vehicleChecks.showMeQuestions instanceof Array) ||
+    data.vehicleChecks.showMeQuestions.length !== REQUIRED_SHOW_ME_QUESTIONS
+   ) {
+    showMeQuestionComplete = false;
+  } else {
+    data.vehicleChecks.showMeQuestions.forEach((element) => {
+      if (
+        !(element.outcome === CompetencyOutcome.P ||
+        element.outcome === CompetencyOutcome.DF ||
+        element.outcome === CompetencyOutcome.S ||
+        element.outcome === CompetencyOutcome.D)
+        ) {
+        showMeQuestionComplete = false;
+      }
+    });
+  }
+
+  if (
+    typeof data.vehicleChecks.tellMeQuestions === 'undefined' ||
+    data.vehicleChecks.tellMeQuestions === null ||
+    !(data.vehicleChecks.tellMeQuestions instanceof Array) ||
```

Comment:
Perhaps a helper method that returns a boolean  for this check to reduce code duplication.

Reply:
Have reduced down to a simplified instanced check